docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,161 @@
+
+```ts
+import {Order} from './order.model.ts';
+import {Entity, property, hasmany} from '@loopback/repository';
+
+export class Customer extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+  })
+  id: number;
+
+  @property({
+    type: 'string',
+  })
+  name: string;
+
+  @hasMany(Order) orders: Order[];
+}
+```
+
+The metadata generated for the above decoration is as follows:
+
+```js
+{
+  type: 2; // denotes hasMany relation (this is an enum value)
+  keyTo: customerId; // Order model should have this property declared
+}
+```
+
+The property type metadata is also preserved as an array of type `Order` as part
+of the decoration. Another usage of the decorator with a custom foreign key name
+for the above example is as follows:
+
+```ts
+// inside Customer model class
+@hasMany(Order, {keyTo: 'custId'}) orders: Order[];
+```
+
+## Configuring a HasMany relation
+
+The configuration and resolution of a `HasMany` relation takes place at the
+repository level. Once `HasMany` relation is defined on the source model, then
+there are a couple of steps involved to configure it and use it. On the source
+repository, the following are required:
+
+- Use [Dependency Injection](Dependency-injection.md) to inject an instance of
+  the target repository in the constructor of your source repository class.
+- Declare a property with the factory function type
+  `HasManyRepositoryFactory<TargetModel, typeof TargetModel.prototype.id>` on
+  the source repository class.
+- call the `_createHasManyRepositoryFactoryFor` function in the constructor of
+  the source repository class with the relation name (decorated relation propert
+  on the source model) and target repository instance and assign it the property
+  mentioned above.
+
+The following code snippet shows how it would look like:
+
+{% include code-caption.html
+content="/src/repositories/customer.repository.ts.ts" %}
+
+```ts
+import {Order, Customer} from '../models';
+import {OrderRepository} from './order.repository.ts';
+import {
+  DefaultCrudRepository,
+  juggler,
+  HasManyRepositoryFactory,
+} from '@loopback/repository';
+
+class CustomerRepository extends DefaultCrudRepository<
+  Customer,
+  typeof Customer.prototype.id
+> {
+  public orders: HasManyRepositoryFactory<Order, typeof Customer.prototype.id>;
+  constructor(
+    @inject('datasources.db') protected db: juggler.DataSource,
+    @repository(OrderRepository) orderRepository: OrderRepository,
+  ) {
+    super(Customer, db);
+    this.orders = this._createHasManyRepositoryFactoryFor(
+      'orders',
+      orderRepository,
+    );
+  }
+}
+```
+
+The following CRUD APIs are now available in the constrained target repository
+factory `orders` for instances of `customerRepository`:
+
+- `create` for creating a target model instance
+- `find` finding target model instance(s)
+- `delete` for deleting target model instance(s)
+- `patch` for patching target model instance(s)
+
```

Comment:
Please add a link pointing to our API documentation for these CRUD API methods. Or perhaps one link for each method?

```markdown
- `create` for creating a target model instance belonging to the source model instance ([API docs](<insert-link>)).
```